vk/router.go

```diff
@@ -139,6 +149,52 @@ func (rt *Router) handlerWrap(inner HandlerFunc) httprouter.Handle {
 	}
 }
 
+// wsHandlerWrap returns an httprouter.Handle that uses the `inner` vk.WebSocketHandleFunc to handle the request
+//
+// inner accepts a Gorilla `Conn` and reads and writes messages to it
+//
+func (rt *Router) wsHandlerWrap(inner WebSocketHandlerFunc) httprouter.Handle {
+	return func(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+		var status int
+		var body []byte
+
+		// create a context handleWrap the configured logger
+		// (and use the ctx.Log for all remaining logging
+		// in case a scope was set on it)
+		ctx := NewCtx(rt.log, params, w.Header())
+		ctx.UseScope(defaultScope{ctx.RequestID()})
+
+		logDone := rt.logRequest(r, ctx)
+
+		upgrader := websocket.Upgrader{
+			ReadBufferSize:  1024,
+			WriteBufferSize: 1024,
+			CheckOrigin:     func(r *http.Request) bool { return true },
```

Comment:
Technically it's not a requirement that WebSocket servers enforce origin checks, it is however strongly encouraged. 

> Servers that are not intended to process input from any web page but
   only for certain sites SHOULD verify the |Origin| field is an origin
   they expect.
   
  https://www.rfc-editor.org/rfc/rfc6455.html#section-10.2

I think disabling the redundant check in the upgrader makes sense so long as you add a comment explaining why its done this way and add documentation stating that we recommend only using this functionality in conjunction with the CORS middleware.

Reply:
CORS won't quite work since that's enforced client-side, but yeah I think middleware is the answer here. Middleware won't be able to reject the connection entirely which would be a bit nicer though.

Reply:
Will mention this in the documentation.

Reply:
Yep this would be something accomplished with the CORS middleware IMO, in fact it won't work cross-origin from a web browser unless the server does add the CORS middlware. From non-browser it's not enforced, and I don't think anyone would expect us to do enforcement.